### PR TITLE
replace 'whitelist' and 'blacklist' language

### PR DESCRIPTION
**Pull request by liewegas**
There is no need to use terms that may be viewed as racially charged.

There are a few compatibility challenges there:

- a few config options are renamed.  These are done without backward compatibility.  The mon will remap them on upgrade.
- a few CLI commands are changed.  The old 'blacklist' variants are deprecated but will still work for now.

TODO
- [x] make mon adjust config options automatically
- [ ] include upgrade test (once initial octopus-x tests merge) that sets old configs to magic values, upgrades, and verifies new config options reflect the same values.

### CODE REVIEW

**Comment by batrick** [comment]
I suppose we'll also be renaming the prolific use of master/slave everywhere too (MDS in particular)?

Big +1 on this effort.

File: PendingReleaseNotes [hunk 1]
```diff
@@ -77,3 +77,28 @@
   librbd to load the plugin by adding the following to your configuration::
 
     rbd_plugins = parent_cache
+
+* `blacklist` has been replaced with `blocklist` throughout.  The following commands have changed:
+
+  - ``ceph osd blacklist ...`` are now ``ceph osd blocklist ...``
```
Comment (LenzGr) [inline PendingReleaseNotes:83]:
```suggestion
  - ``ceph osd blacklist ...`` is now ``ceph osd blocklist ...``
```

**Comment by Devp00l** [comment]
@batrick from [‘Master/Slave’ Terminology Was Removed from Python Programming Language](https://www.vice.com/en_us/article/8x7akv/masterslave-terminology-was-removed-from-python-programming-language)
> In the end, most of Stinner’s suggested terminology changes were adopted. Thus, “slaves” was changed to “workers” or “helpers” and “master process” to “parent process.”

**Comment by asettle** [comment]
@liewegas are you planning on backporting? And if so, how far back? :)

**Comment by socketpair** [comment]
I would vote against if it was possible.

**Comment by freehck** [comment]
> I would vote against if it was possible.
I agree. Thumbs down.
Please don't implement Orwell's Newspeak.

File: PendingReleaseNotes (hunk 1 above)
Comment (liewegas) [reply PendingReleaseNotes:83]: I did 'are' because the '...' refers to a bunch of commands.

**Comment by batrick** [comment]
> > I would vote against if it was possible.
> 
> I agree. Thumbs down.
> Please don't implement Orwell's Newspeak.

I'm glad you brought up Orwell, here would be his thoughts on the subject;

"But if thought corrupts language, language can also corrupt thought. A bad usage can spread by tradition and imitation, even among people who should and do know better. [...]

[...] Political language – and with variations this is true of all political parties, from Conservatives to Anarchists – is designed to make lies sound truthful and murder respectable, and to give an appearance of solidity to pure wind. One cannot change this all in a moment, but one can at least change one’s own habits, and from time to time one can even, if one jeers loudly enough, send some worn-out and useless phrase – some jackboot, Achilles’ heel, hotbed, melting pot, acid test, veritable inferno or other lump of verbal refuse – into the dustbin where it belongs."

https://www.orwellfoundation.com/the-orwell-foundation/orwell/essays-and-other-works/politics-and-the-english-language/

**Comment by asettle** [comment]
I'm not sure "blocklist" and "ignorelist" are the clearest options we could work with. Here are some comments from myself and colleagues:

- "Ignore" can be interpreted as both "ignore this, let them in" or "ignore this, do not let them in".
- To use 'ignorelist' as a verb is strange, we would recommend 'allow'. To use 'ignorelist' as a noun has a wrong meaning, one can think
that 'ignorelisted' members will be ignore, when the contrary is true
- "Blocklist" is clear but sounds a lot like "blockchain". Although "blocklisting" something is an odd turn of phrase. We are simply blocking the user from access, yes?

Interested in other thoughts/opinions.

**Comment by socketpair** [comment]
"A Ceph Monitor maintains a master copy of the cluster map." and also "master/slave OSDs. What you think about renaming them too?

**Comment by krakazyabra** [comment]
I also suggest to remove white color from WebUI. White color dominates in interface, so black people can think, that whites masters dominate.
Add more brown, all shades of brown. Background should be black. Same as font.

**Comment by jan--f** [comment]
> I'm not sure "blocklist" and "ignorelist" are the clearest options we could work with.

Iiuc the change to ignorelist is only done for the qa job specs, indicating message that can be safely ignored. So ignorelist works well in this context.

> Interested in other thoughts/opinions.

accept/deny would also be a potential substitute for allow/block.

**Comment by vitalif** [comment]
I partly understand native English speakers when they discuss the removal of master/slave terminology - after all, in Russian it's never been called master/slave literally, the same thing was always called primary/secondary or driver/driven.
However, replacing blacklist and whitelist seems like a definite overkill. Black and white colours have never been connected with skin. It was just the general "darkness is frightening" coming from the ancient ages I suppose.
Also I think that the enthusiasm with which everyone started renaming terms is scary. Don't you think this can lead to unwanted social situations where nobody will be allowed to say anything he wants to say just because there exists some person who doesn't like it?
And also +1 to the `ignorelist` not being a clear synonym for `whitelist`.

**Comment by batrick** [comment]
> I partly understand native English speakers when they discuss the removal of master/slave terminology - after all, in Russian it's never been called master/slave literally, the same thing was always called primary/secondary or driver/driven.
> However, replacing blacklist and whitelist seems like a definite overkill. Black and white colours have never been connected with skin. It was just the general "darkness is frightening" coming from the ancient ages I suppose.

That may be the legitimate origin of the phrase but that doesn't mean it should survive the purge. The overall point is that language is an insidious mechanism for inertia of power, in this case that white people have very real advantages over black people. Even if the term does not have racist roots, it helps perpetuate the idea that color matters at some level. As a group, we choose to reject that by finding alternate (and clearer) terms.

> Also I think that the enthusiasm with which everyone started renaming terms is scary.

Awareness means more than just acknowledging the pains of others. Change is required. Whether this is something like a police officer choosing to not question a black teenager on campus about whether he has a student identity card or engineers purging  terminology from interfaces/manuals that reinforces racist mindsets

> Don't you think this can lead to unwanted social situations where nobody will be allowed to say anything he wants to say just because there exists some person who doesn't like it?

I like to think that people are more forgiving of what is said off-hand than what is written down. It's decidedly hard to change habits of speech. Choose to educate rather than criticize. Practice forgiveness.

**Comment by liewegas** [comment]
> I partly understand native English speakers when they discuss the removal of master/slave terminology - after all, in Russian it's never been called master/slave literally, the same thing was always called primary/secondary or driver/driven.
> However, replacing blacklist and whitelist seems like a definite overkill. Black and white colours have never been connected with skin. It was just the general "darkness is frightening" coming from the ancient ages I suppose.

Perhaps 'blacklist' and 'whitelist' do not have racial overtones in Russia, but here in the US, in this moment, they most definitely do.

> Also I think that the enthusiasm with which everyone started renaming terms is scary. Don't you think this can lead to unwanted social situations where nobody will be allowed to say anything he wants to say just because there exists some person who doesn't like it?

No. I am less concerned with people being afraid of the suggestion that they might be inadvertantly using racially-charged language than the systems of racism that reinforce the use of such language and resist any change--however painless--to retire it.  People can still use whatever words they like, but as a project and community, we are not obliged to use racially-loaded terms in our software, or to tolerate it in our technical discussions.  If nothing else, simple steps like this PR are a statement that the developer community cares enough about inclusiveness and racial justice to endure the trivial inconvenience of renaming a few variables and commands, and the acknowledge that we can always learn and do better.

**Comment by kshtsk** [comment]
Alternatively there can be used: include/exclude pairs.

**Comment by vitalif** [comment]
> Practice forgiveness

I prefer to practice directness :-)
Yeah, it's really hard to understand the current situation in the US from Russia so these renamings look as a bit of circus to most Russian people. You probably know the situation better, so you decide, anyway :)

**Comment by socketpair** [comment]
Is there any black guy who is really annoyed/upset by these terms ? I think no. It's pure circus. Pure hype.

**Comment by freehck** [comment]
> > Please don't implement Orwell's Newspeak.
> 
> I'm glad you brought up Orwell, here would be his thoughts on the subject;

I find it's strange, that somebody citing such things votes _for_, not _against_. Did you understand what Orwell wrote?

He wrote that language influences thoughts. Because our language defines what we can think about. Therefore it's definitely not a good idea to enforce somebody to use any kind of "political language". Enforcement can't be the right way.

I vote against, and ask the project maintainers again: please don't do it.

**Comment by socketpair** [comment]
Для русскоговорящих: Скоро будем вместо "чёрная дыра" говорить "афроотверстие"

**Comment by Devp00l** [comment]
Hm... there seems to be a lot of misunderstandings about terminology here.

In this case it's about the change from "whitelist" and "blacklist", which clearly separates white listed entries from black listed entries. White is allowed, black is denied or needs to be filtered.
Now read the last sentence and ask anyone if this sentence sounds a bit odd.

We create open source here in order that anyone anywhere on the planet can use it and contribute to it. Showing that we care about a comfortable language for everyone makes the use and development easier for everyone in the end, even if one doesn't see it, because he or she is not affected by those words. If you are not affected and don't know anybody who is by some words used in daily language, you are lucky.

It's also not about any hype, it's simply just the right time to actively think about about modernizing the language in use (everywhere) and these thoughts will not go away. For example discussions and changes about wordings is a common practice with a lot of great child books which should include modern terms now replacing the old terms, that were ok in the time the book was created or updated but aren't ok anymore.

**Comment by fullerdj** [comment]
We tend to use this in a verb context a lot; any interest in fance/fenced?

**Comment by freehck** [comment]
> a comfortable language for everyone

It's obviously impossible.

**Comment by Devp00l** [comment]
> > a comfortable language for everyone
> 
> It's obviously impossible.

It's just development overall it's WIP ;) and some time words get deprecated.

**Comment by markhpc** [comment]
> We tend to use this in a verb context a lot; any interest in fence/fenced and removing the 'list' part?

I'd vote for allowed/denied and allowedlist/deniedlist myself.

**Comment by neha-ojha** [comment]
superseded by https://github.com/ceph/ceph/pull/36527

**Pull request closed without merging**